[core] Rez plugins: Ensure dependencies for `meshroom_compute` are present

cbentejac opened the pull request:
## Description

This PR introduces a mechanism to ensure that required dependencies for `meshroom_compute`, specifically `psutil`, are always included in the node execution environment for rez plugins. Dependencies for `meshroom_compute` are added to the environment without specifying any version or attempting to match Meshroom's, in order for it to be set directly during the environment resolution.

**Dependency management improvements:**

* Introduced a new constant `_MESHROOM_COMPUTE_DEPS` in `meshroom/core/desc/node.py` to define required dependencies for `meshroom_compute`.
* Updated `meshroom/core/plugins.py` to import `_MESHROOM_COMPUTE_DEPS` and ensure all dependencies listed in it (currently `psutil`) are always included in the resolved execution environment for plugins, unless already present [[1]](diffhunk://#diff-1d5f2a46e59213721e5624ac787e491e5cc2333ab8838b226c6558ceb0d87913L17-R17) [[2]](diffhunk://#diff-1d5f2a46e59213721e5624ac787e491e5cc2333ab8838b226c6558ceb0d87913R197-R205).

fabiencastan commented on meshroom/core/plugins.py:
```diff
@@ -194,6 +194,15 @@ def resolveRezSubrequires(self) -> list[str]:
+            # the environment based on the other packages.
+            # If any of these packages is already part of the environment a plugin's dependency,
+            # do not add it
+            if not package in " ".join(packages):
```
```suggestion
            if package not in " ".join(packages):
```

fabiencastan commented on meshroom/core/plugins.py:
```diff
@@ -194,6 +194,15 @@ def resolveRezSubrequires(self) -> list[str]:
                 if not definedInParentEnv:
                     packages.append(package)
 
+        for package in _MESHROOM_COMPUTE_DEPS:
+            # For packages that are required by meshroom_compute, do not specify any version
+            # or align it with Meshroom's: the version will be found during the resolution of
+            # the environment based on the other packages.
+            # If any of these packages is already part of the environment a plugin's dependency,
+            # do not add it
+            if not package in " ".join(packages):
+                packages.append(package)
```
```suggestion
        def extract_package_name(package_string):
            return re.split(r'[-=<>!~]+', package_string, 1)[0]
        package_names = [extract_package_name(pkg) for pkg in packages]

        for package in _MESHROOM_COMPUTE_DEPS:
            # For packages that are required by meshroom_compute, do not specify any version
            # or align it with Meshroom's: the version will be found during the resolution of
            # the environment based on the other packages.
            # If any of these packages is already part of the environment a plugin's dependency,
            # do not add it
            if package not in package_names :
                packages.append(package)
```

cbentejac replied:
Done

cbentejac committed:
[core] Rez plugins: Ensure dependencies for `meshroom_compute` are present

`psutil` is a dependency of `meshroom_compute`, which cannot run if it
is not part of the environment. For rez plugins, if `psutil` is not
already part of the plugin's environment, it is added. No version is
specified so it can be resolved in the environment, without being aligned
on Meshroom's.

fabiencastan committed:
[core] plugins: use REZ_DELIMITER_PATTERN
